torch/_inductor/compile_fx.py

```diff
@@ -693,7 +694,55 @@ def fw_compiler_base(model: torch.fx.GraphModule, example_inputs, is_inference):
         )
 
     fw_compiler = functools.partial(fw_compiler_base, is_inference=False)
-    inference_compiler = functools.partial(fw_compiler_base, is_inference=True)
+
+    if config.freezing:
+        from torch._inductor.freezing import freeze
+
+        def inference_compiler(
+            aot_autograd_model: torch.fx.GraphModule, example_inputs
+        ):
+            # partition_fn won't be called
+            joint_graph_passes(aot_autograd_model)
+
+            opt_model, preserved_arg_indices = freeze(
+                model_,
+                aot_autograd_model,
+                fw_metadata=torch._guards.TracingContext.get().fw_metadata,
+            )
+
+            example_inputs = [example_inputs[ind] for ind in preserved_arg_indices]
+            num_fixed = len(preserved_arg_indices) - num_example_inputs
+
+            fake_mode = detect_fake_mode(example_inputs)
+
+            # constant params will be real tensors, not fake
+            with unittest.mock.patch.object(fake_mode, "allow_non_fake_inputs", True):
+                optimized_function = inner_compile(
+                    opt_model,
+                    example_inputs,
+                    num_fixed=num_fixed,
+                    cudagraphs=cudagraphs,
+                    graph_id=graph_id,
+                    is_inference=True,
+                    boxed_forward_device_index=forward_device,
+                )
+
+            # Need to drop the args we have constant-ified.
+            params_flat = torch._guards.TracingContext.get().params_flat
```

Comment:
Maybe this should happen inside of `freeze()` too, so all of the calling convention changes happen in one place?